momento-sdk/src/main/java/momento/sdk/ScsTopicClient.java

```diff
@@ -0,0 +1,134 @@
+  private final CredentialProvider credentialProvider;
+
+  public ScsTopicClient(
+      @Nonnull CredentialProvider credentialProvider, @Nonnull TopicConfiguration configuration) {
+    this.credentialProvider = credentialProvider;
+    this.topicGrpcStubsManager = new ScsTopicGrpcStubsManager(credentialProvider, configuration);
+  }
+
+  public CompletableFuture<TopicPublishResponse> publish(
+      String cacheName, String topicName, byte[] value) {
+    try {
+      ValidationUtils.checkCacheNameValid(cacheName);
+      ValidationUtils.checkTopicNameValid(topicName);
+      ValidationUtils.ensureValidValue(value);
+    } catch (Exception e) {
+      return CompletableFuture.completedFuture(
+          new TopicPublishResponse.Error(CacheServiceExceptionMapper.convert(e)));
+    }
+
+    _TopicValue topicValue = _TopicValue.newBuilder().setBinary(ByteString.copyFrom(value)).build();
+    return sendPublish(cacheName, topicName, topicValue);
+  }
+
+  public CompletableFuture<TopicPublishResponse> publish(
+      String cacheName, String topicName, String value) {
+    try {
+      ValidationUtils.checkCacheNameValid(cacheName);
+      ValidationUtils.checkTopicNameValid(topicName);
+      ValidationUtils.ensureValidValue(value);
+    } catch (Exception e) {
+      return CompletableFuture.completedFuture(
+          new TopicPublishResponse.Error(CacheServiceExceptionMapper.convert(e)));
+    }
+
+    _TopicValue topicValue = _TopicValue.newBuilder().setText(value).build();
+    return sendPublish(cacheName, topicName, topicValue);
+  }
+
+  public CompletableFuture<TopicSubscribeResponse> subscribe(
+      String cacheName, String topicName, ISubscriptionCallbacks options) {
+    try {
+      ValidationUtils.checkCacheNameValid(cacheName);
+      ValidationUtils.checkTopicNameValid(topicName);
+    } catch (Exception e) {
+      return CompletableFuture.completedFuture(
+          new TopicSubscribeResponse.Error(CacheServiceExceptionMapper.convert(e)));
+    }
+    return sendSubscribe(cacheName, topicName, options);
+  }
+
+  private CompletableFuture<TopicPublishResponse> sendPublish(
+      String cacheName, String topicName, _TopicValue value) {
+    CompletableFuture<TopicPublishResponse> future = new CompletableFuture<>();
+
+    _PublishRequest request =
+        _PublishRequest.newBuilder()
+            .setCacheName(cacheName)
+            .setTopic(topicName)
+            .setValue(value)
+            .build();
+
+    try {
+      topicGrpcStubsManager
+          .getStub()
+          .publish(
+              request,
+              new StreamObserver() {
+
+                @Override
+                public void onNext(Object value) {
+                  // Do nothing
+                }
+
+                @Override
+                public void onError(Throwable t) {
+                  future.completeExceptionally(
+                      new TopicPublishResponse.Error(CacheServiceExceptionMapper.convert(t)));
```

Comment:
I think that if you want the end user to get an instance of `Error` without it being thrown as an exception, you probably don't want to call `completeExceptionally` here? I can test it again with that program that I pasted that intentionally causes throttles. or we can test with a bad cache name - I can't remember what is supposed to happen in other SDKs if you do `publish` with a cache name that doesn't exist. If that returns an `Error` response type in the other SDKs then maybe we should add a test for that case in your code, to verify how these completes are working.